src/copilot/generateOpenApi.ts

```diff
@@ -0,0 +1,191 @@
+${await getRuleDescriptions(rulesetFile)}`,
+        };
+
+        const chatResponseParts: vscode.ChatResponsePart[] = [];
+
+        chatResponseParts.push(new vscode.ChatResponseMarkdownPart(UiStrings.GenerateOpenApiRegisterApiDesc));
+        chatResponseParts.push(new vscode.ChatResponseCommandButtonPart({
+            title: UiStrings.GenerateOpenApiRegisterApiButton,
+            command: "azure-api-center.registerApi",
+            arguments: []
+        }));
+
+        if (rulesetFile) {
+            chatResponseParts.push(new vscode.ChatResponseMarkdownPart(UiStrings.GenerateOpenApiViewApiDesc));
+            chatResponseParts.push(new vscode.ChatResponseCommandButtonPart({
+                title: UiStrings.GenerateOpenApiViewApiButton,
+                command: "workbench.view.extension.api-center-treeview",
+                arguments: []
+            }));
+        } else {
+            chatResponseParts.push(new vscode.ChatResponseMarkdownPart(UiStrings.GenerateOpenApiSetRuleDesc));
+            chatResponseParts.push(new vscode.ChatResponseCommandButtonPart({
+                title: UiStrings.GenerateOpenApiSetRuleButton,
+                command: "azure-api-center.setApiRuleset",
+                arguments: []
+            }));
+
+            chatResponseParts.push(new vscode.ChatResponseMarkdownPart(UiStrings.GenerateOpenApiRegenerateDesc));
+            chatResponseParts.push(new vscode.ChatResponseCommandButtonPart({
+                title: UiStrings.GenerateOpenApiRegenerateButton,
+                command: "workbench.action.chat.open",
+                arguments: [{
+                    query: `@azure ${agentRequest.userPrompt}`,
+                }]
+            }));
+        }
+
+        return {
+            responseForLanguageModel: responseForLanguageModel,
+            chatResponseParts: chatResponseParts
+        };
+    }
+
+    function getRulesetFile(): string | undefined {
+        const config = vscode.workspace.getConfiguration('spectral');
+        const rulesetFile = config.get<string>('rulesetFile');
+        return rulesetFile;
+    }
+
+    async function getRuleDescriptions(rulesetFile: string | undefined): Promise<string> {
+        if (rulesetFile) {
+            const rulesetFileContent = await getRulesetFileContent(rulesetFile);
+            if (!rulesetFileContent) {
+                return "";
+            }
+
+            const ruleDescriptions = getRuleDescriptionsFromRulesetFileContent(rulesetFileContent);
+            return ruleDescriptions;
+        } else {
+            return spectralDefaultRuleDescriptions;
+        }
+    }
+
+    async function getRulesetFileContent(rulesetFile: string): Promise<string | undefined> {
+        try {
+            let rulesetFileContent = "";
+            if (rulesetFile.startsWith('http://') || rulesetFile.startsWith('https://')) {
+                rulesetFileContent = await axios.get(rulesetFile).then(response => response.data);
```

Comment:
It's mixing await/async and promise resolution with .then()/.catch(), so just pick one.